### PR TITLE
Develop: general updates from my work branch gema_work

### PR DESCRIPTION
**Pull request by gpvillegas**
## Test runs used
All these preliminary calibrations were done using the following test runs (1 for each kinematic).

| 20840  | heep_coin|
| ------------- | ------------- |
| SHMS 3/4 (PS1)  | -1  |
| SHMS-ELREAL (PS2)  | -1  |
| HMS 3/4 (PS3)  | -1  |
| HMS-ELREAL (PS4)  | -1  |
| SHMS-ELREAL x HMS 3/4  (PS5)  | -1  |
| SHMS 3/4 x HMS 3/4 (PS6)  | 0  |

| 20870  | heep_singles|
| ------------- | ------------- |
|  PS1  |  0  |
|  PS2  | -1  |
|  PS3  | -1  |
|  PS4  | -1  |
|  PS5  | -1  |
|  PS6  |  0  |

| 20871  | heep_singles|
| ------------- | ------------- |
|  PS1  |  -1 |
|  PS2  | -1  |
|  PS3  | -1  |
|  PS4  | -1  |
|  PS5  | -1  |
|  PS6  |  0  |

| 21049  | lumi |
| ------------- | ------------- |
|  PS1  | -1  |
|  PS2  |  0  |
|  PS3  | -1  |
|  PS4  | -1  |
|  PS5  | -1  |
|  PS6  | -1  |

## Main changes
1. Reftimes set for SHMS, HMS detectors in their respective `.param` files.
2. Changed workflow for setting reftime for singles runs (experimental)
    - Since deuteron only ran singles on SHMS, I created `p_reftime_singles_cut.param` with the appropriate parameters,   though I am still debating whether this is really needed.
    - Now for selected singles runs on `standard.database` you will need to use the `general_deut_singles_spring23.param` as the general parameter file where the reftime cut parameter files are summarized for the code.
3. Time window cuts were set based on the new cal and hodo parameters `set_timewin.sh` gives, then tweaked manually for all test runs.

## Pending for pass 0.5
1. `tcoin.param` for TRIG pseudo-detectors has not been set yet (working on it)

### CODE REVIEW

File: CALIBRATION/set_reftimes/scripts/set_reftimes.C
Comment (gpvillegas) [inline CALIBRATION/set_reftimes/scripts/set_reftimes.C:1]: here I made most changes to have a separate param file for singles runs.

File: CALIBRATION/set_reftimes/set_reftime.sh
Comment (gpvillegas) [inline CALIBRATION/set_reftimes/set_reftime.sh:1]: added option for `set_reftime.sh` script to use different param file for singles run

File: DBASE/COIN/GEN/general_deut_singles_spring23.param
Comment (gpvillegas) [inline DBASE/COIN/GEN/general_deut_singles_spring23.param:1]: new general param file with p_reftime_singles param file

**Review by gpvillegas — COMMENTED** [review]
checked over all new files added and changes I made, from now on ill try to push important changes more often so that they don't get backed up like this, I decided to leave root files and histogram files I made while replaying the data for others to use.

**Review by Yero1990 — COMMENTED** [review]
Remember, we should not merge changes that involve a file path change as that may alter other users, for example, if one tries to run a replay on ifarm, but the CACHE links is changed to a local path.

**Review by Yero1990 — COMMENTED** [review]
Remember, changes like setting a local file path: 
 const char* ROOTFileNamePattern = "/home/gvill/deuteron/ROOTfiles/%s/deut_replay_%s_%d_%d.root";
in replay_deut.C can affect other user when they try to do a replay.

**Review by Yero1990 — COMMENTED** [review]
Remember,  local changes to file paths should not be merged. Perhaps can be left as a general file path, like finding what is ${analysis_home}, depending on the user

**Review by Yero1990 — COMMENTED** [review]
OK, I see you reverted the changes in another commit. OK thanks. Yes, GitHub is a bit confusing. Perhaps, we should find a way to modify the file paths such that it looks
for a user using raw ${PWD} command to display the user current directory and then
go from there, that way file paths could be kept general. I don't know if this is easy.

**Comment by gpvillegas** [comment]
I think its a good idea, I'll try it and then push it. I also realized I merged all the 25 different commits but github gives you an option to merge&squash so that it looks only like 1 so Ill do that instead.

**Pull request merged**